core/domain/wipeout_service.py

```diff
@@ -1008,19 +1084,15 @@ def _pseudonymize_models_transactional(
                 ]
             rights_snapshot_content_model.content = model_dict
             rights_snapshot_content_model.update_timestamps()
+        datastore_services.put_multi(rights_snapshot_content_models)
 
         commit_log_models = [
             model for model in activity_related_models
             if isinstance(model, commit_log_model_class)]
         for commit_log_model in commit_log_models:
             commit_log_model.user_id = pseudonymized_id
             commit_log_model.update_timestamps()
-
-        datastore_services.put_multi(
-            snapshot_metadata_models +
-            rights_snapshot_metadata_models +
-            rights_snapshot_content_models +
-            commit_log_models)
+        datastore_services.put_multi(commit_log_models)
```

Comment:
Same notes as above re not splitting to avoid an inconsistent state; also, one call is faster than three.

In general, put_multi() _should_ be able to accept multiple kinds of models, so probably worth digging more into why mypy can't handle this. /cc @vojtechjelinek

Reply:
As mentioned above in the comment https://github.com/oppia/oppia/pull/15827#discussion_r945132048 I have added a Ignore here